runtime/src/main/kotlin/kotlin/native/Text.kt

```diff
@@ -8,40 +8,48 @@ package kotlin.native
 /**
  * Converts an UTF-8 array into a [String]. Replaces invalid input sequences with a default character.
  */
-public fun ByteArray.stringFromUtf8(start: Int = 0, size: Int = this.size) : String =
-        stringFromUtf8Impl(start, size)
+public fun ByteArray.stringFromUtf8(start: Int = 0, size: Int = this.size) : String {
+    AbstractList.checkBoundsIndexes(start, start + size, this.size)
```

Comment:
What's that? Why not top level function?

Reply:
It's an internal function in the common standard library for bound checking, logically scoped to the companion object of `AbstractList` class. Does it present a problem for Kotlin/Native?

Reply:
Using companion here produced unneeded slowdown, as we need to perform init check. Please use top level here.